overlord/devicestate/handlers_install.go

```diff
@@ -76,16 +84,27 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 			// enable data encryption
 			"--encrypt",
 			// location to store the keyfile
-			"--key-file", filepath.Join(boot.InitramfsUbuntuBootDir, "ubuntu-data.keyfile.unsealed"),
+			"--key-file", filepath.Join(boot.InitramfsEncryptionKeyDir, "ubuntu-data.keyfile.sealed"),
+			// location to store the recovery keyfile
+			"--recovery-key-file", filepath.Join(boot.InitramfsWritableDir, "var/lib/snapd/device/fde/recovery-key"),
 			// location to store the recovery keyfile
-			"--recovery-key-file", filepath.Join(boot.InitramfsUbuntuDataDir, "/system-data/var/lib/snapd/device/fde/recovery-key"),
+			"--recovery-key-file", filepath.Join(boot.InitramfsWritableDir, "recovery-key"),
+			// location to store the lockout authorization data
+			"--lockout-auth-file", filepath.Join(boot.InitramfsWritableDir, "lockout-auth"),
+			// location to store the authorization policy update data
+			"--auth-update-file", filepath.Join(boot.InitramfsWritableDir, "auth-update"),
+			// path to the kernel to install
+			"--kernel", filepath.Join(kernelDir, "kernel.efi"),
```

Comment:
So when we are sealing this, we are just providing a path to the kernel to be read and sealed against right now, not the path that the kernel will actually be booted at, right? Because if this path is supposed to be the path that the kernel is booted from, then this is wrong, and it should actually be something like `/ubuntu/kernel.efi` (which is the path of the kernel.efi we boot from ubuntu-boot when grub boots the kernel).

Reply:
Yes, this is the path to the kernel that will be used in the sealing process.

Reply:
thanks for the explanation